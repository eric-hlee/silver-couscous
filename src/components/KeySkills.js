import React from 'react';
import Card from 'react-bootstrap/Card';
import ListGroup from 'react-bootstrap/ListGroup';

export default function KeySkills() {
  const getSkillItem = (skill, index) => {
    return <ListGroup.Item key={index}>{skill}</ListGroup.Item>;
  };

  return (
    <Card className="key-skills">
      <Card.Body>
        <Card.Text as="div">
          <Card.Title>Programming languages:</Card.Title>
          <ListGroup horizontal="sm">
            {['Python', 'Java', 'Bash'].map(getSkillItem)}
          </ListGroup>
          <br />
          <Card.Title>Web/scripting languages:</Card.Title>
          <ListGroup horizontal="sm">
            {['HTML', 'CSS', 'Sass', 'JavaScript', 'SQL'].map(getSkillItem)}
          </ListGroup>
          <br />
          <Card.Title>Library/Frameworks:</Card.Title>
          <ListGroup horizontal="sm">
            {[
              'React',
              'Redux',
              'React-Router',
              'React Intl',
              'Node.js',
              'Draft.js',
              'Jest',
              'Enzyme',
            ].map(getSkillItem)}
          </ListGroup>
          <br />
          <Card.Title>Other technologies:</Card.Title>
          <ListGroup horizontal="sm">
            {['Docker', 'Git', 'GitHub', 'Webpack'].map(getSkillItem)}
          </ListGroup>
        </Card.Text>
      </Card.Body>
    </Card>
  );
}
