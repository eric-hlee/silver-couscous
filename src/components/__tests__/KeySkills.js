import React from 'react'
import { shallow } from 'enzyme'
import KeySkills from '../KeySkills'

describe('<KeySkills />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<KeySkills />)
    expect(wrapper).toMatchSnapshot()
  })
})
