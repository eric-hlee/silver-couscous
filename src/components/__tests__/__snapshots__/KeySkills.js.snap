// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeySkills /> renders correctly 1`] = `
<Card
  body={false}
  className="key-skills"
>
  <CardBody>
    <CardText
      as="div"
    >
      <CardTitle>
        Programming languages:
      </CardTitle>
      <ListGroup
        horizontal="sm"
        variant={null}
      >
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Python
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Java
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Bash
        </ListGroupItem>
      </ListGroup>
      <br />
      <CardTitle>
        Web/scripting languages:
      </CardTitle>
      <ListGroup
        horizontal="sm"
        variant={null}
      >
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          HTML
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          CSS
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Sass
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          JavaScript
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          SQL
        </ListGroupItem>
      </ListGroup>
      <br />
      <CardTitle>
        Library/Frameworks:
      </CardTitle>
      <ListGroup
        horizontal="sm"
        variant={null}
      >
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          ReactJS
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Redux
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          React-Router
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          React Intl
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          ExpressJS
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          DraftJS
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Jest
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Enzyme
        </ListGroupItem>
      </ListGroup>
      <br />
      <CardTitle>
        Other technologies:
      </CardTitle>
      <ListGroup
        horizontal="sm"
        variant={null}
      >
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Docker
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Git
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          GitHub
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          variant={null}
        >
          Webpack
        </ListGroupItem>
      </ListGroup>
    </CardText>
  </CardBody>
</Card>
`;
